Tests/MathLib/TestGlobalVectorInterface.cpp

```diff
@@ -187,6 +189,75 @@ void checkGlobalVectorInterfaceMPI()
     MathLib::finalizeVectorAssembly(x_fixed_p);
     T_VECTOR x_deep_copied(x_fixed_p);
     ASSERT_NEAR(sqrt(3.0*5), x_deep_copied.getNorm(), 1.e-10);
+
+    // -----------------------------------------------------------------
+    // Vector with ghost entries
+    /*
+         Assume there is a vector distributed over three processes as
+          -- rank0 --    --- rank1 ---   -- rank2 --
+           0  1  2  3    4  5  6  7  8   9   10   11 
+         where the numbers are the global entry indices.
+         In each trunk of entries of a rank, there are ghost entries in
+         other ranks attached and their global entry indices are:
+         rank0: 6 8 10
+         rank1: 0 9
+         rank2: 3 5
+
+         Assuming the values of the entries are just their global indices,
+         we have local arrays as:
+         rank0: 0 1 2 3     6 8 10
+         rank1: 4 5 6 7 8   0 9
+         rank2: 9 10 11     3 5
+
+         The above ghost entry embeded vector is realized by the following
+         test.
+    */    
+    std::size_t local_vec_size = 4;
+    if (mrank == 1)
+        local_vec_size = 5;
+    else if (mrank == 2)
+        local_vec_size = 3;
+    std::vector<GlobalIndexType> non_ghost_ids(local_vec_size);
+    std::vector<double> non_ghost_vals(local_vec_size);
+    std::size_t nghosts = 3;
+    if (mrank)
+        nghosts = 2;    
+    std::vector<GlobalIndexType> ghost_ids(nghosts);
+    std::vector<double> expected;
+    switch (mrank)
+    {
+        case 0:
+            non_ghost_ids  = {0, 1, 2, 3};
+            non_ghost_vals = {0., 1., 2., 3.};
+            ghost_ids      = {6, 8, 10};
+            expected       = {0., 1., 2., 3., 6., 8., 10.};
+            break;
+        case 1:
+            non_ghost_ids  = {4, 5, 6, 7, 8};
+            non_ghost_vals = {4., 5., 6., 7., 8.};
+            ghost_ids      = {0, 9};
+            expected       = {4., 5., 6., 7., 8., 0., 9.};
+            break;
+        case 2:
+            non_ghost_ids  = {9, 10, 11};
+            non_ghost_vals = {9., 10., 11.};
+            ghost_ids      = {3, 5};
+            expected       = {9., 10., 11., 3., 5.};
+            break;
+    }
+    T_VECTOR x_with_ghosts(local_vec_size, ghost_ids, is_global_size);
+    x_with_ghosts.set(non_ghost_ids, non_ghost_vals);
+    MathLib::finalizeVectorAssembly(x_with_ghosts);
+
+    ASSERT_EQ(12u, x_with_ghosts.size());
+
+    std::vector<double> loc_v1(  x_with_ghosts.getLocalSize()
+                               + x_with_ghosts.getGhostSize() );
+    x_with_ghosts.getValues(&loc_v1[0]);                          
+    for (std::size_t i=0; i<expected.size(); i++)
+    {
+         ASSERT_NEAR(expected[i], loc_v1[i], 1.e-10);
```

Comment:
Those small integers sould be exactly representable as double values. So `EXPECT_EQ` would be more appropriate, I think. Or is there some argument against it?